flink-table-store-flink/flink-table-store-flink-common/src/main/java/org/apache/flink/table/store/connector/FlinkCatalog.java

```diff
@@ -214,6 +221,53 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
             catalogTable = catalogTable.copy(options);
         }
 
+        if (options.containsKey(CoreOptions.PRIMARY_KEY.key())) {
```

Comment:
These logical can be in `SchemaManager.createTable`? Spark dose not have these codes to take care partition.